apps/web/pages/api/download-cal-video-recording.ts

```diff
@@ -0,0 +1,37 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
+
+import { fetcher } from "@calcom/app-store/dailyvideo/lib/VideoApiAdapter";
+import { getSession } from "@calcom/lib/auth";
+import { defaultHandler, defaultResponder } from "@calcom/lib/server";
+
+type Response = {
+  download_link?: string | undefined;
+  expires?: number;
+  error?: string;
+};
+
+const requestQuery = z.object({
+  recordingId: z.string(),
+});
+
+async function handler(req: NextApiRequest, res: NextApiResponse<Response>) {
```

Comment:
@zomars, should this be under tRPC?

Reply:
@joeauyeung  i am getting the download link only after user has clicked on download ( didn't want to send too many api requests and slow down the experience).  we can easily call this  next js api route to get the link.